src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -319,11 +320,13 @@ private void topControl_MouseLeftButtonDown(object sender, MouseButtonEventArgs
         {
             if (ViewModel == null) return;
 
-            var view = WpfUtilities.FindUpVisualTree<DynamoView>(this);
-
-            ViewModel.DynamoViewModel.ReturnFocusToSearch();
-
-            view.mainGrid.Focus();
+            if ((ViewModel.NodeLogic is CodeBlockNodeModel) == false)
+            {
+                // Do not return focus to search if this is a code block node.
+                var view = WpfUtilities.FindUpVisualTree<DynamoView>(this);
+                ViewModel.DynamoViewModel.ReturnFocusToSearch();
+                view.mainGrid.Focus();
+            }
```

Comment:
Come to think about it, I do agree with you about this being yucky. *Keeping focus* is a UI-specific concern that should be specifiable from a `NodeView`, perhaps through some sort of property like `KeepInputFocusWhenDragged` on `INodeViewCustomization`.